text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,599 @@
+* `mixins` - a list of names that match mixins provided by this buildpack, or the `*` representing all mixins.
+
+## Build Plan (TOML)
+
+```
+[[provides]]
+name = "<dependency name>"
+
+[[requires]]
+name = "<dependency name>"
+mixins = [ "<mixin name>" ]
+
+[requires.metadata]
+# buildpack-specific data; not allowed for mixins
+
+[[or]]
+
+[[or.provides]]
+name = "<dependency name>"
+
+[[or.requires]]
+name = "<dependency name>"
+mixins = [ "<mixin name>" ]
+
+[or.requires.metadata]
+# buildpack-specific data; not allowed for mixins
+```
+
+### Buildpack Plan (TOML)
+
+```
+[[entries]]
+name = "<dependency name>"
+mixin = "<mixin name>"
+
+[entries.metadata]
+# buildpack-specific data
+```
+
+## Lifecycle Interface
+
+### Usage
+
+#### `detector`
+The platform MUST execute `detector` in the **build environment**
+
+Usage:
+```
+/cnb/lifecycle/detector \
+  ...
+  [-stack-buildpacks <stack-buildpacks>] \
+  [-stack-group <group>] \
+  [-stack-order <order>]
+```
+
+##### Inputs
+| Input                 | Environment Variable        | Default Value           | Description
+|-----------------------|-----------------------------|-------------------------|----------------------
+| ...                   |                             |                         |
+| `<stack-buildpacks>`  | `CNB_STACK_BUILDPACKS_DIR`  | `/cnb/stack/buildpacks` | Path to stack buildpacks directory (see [Buildpacks Directory Layout]
+| `<stack-group>`       | `CNB_STACK_GROUP_PATH`      | `./stack-group.toml`    | Path to output group definition(#buildpacks-directory-layout))
+| `<stack-order>`.      | `CNB_STACK_ORDER_PATH`.     | `/cnb/stack/order.toml`| Path to order definition (see order.toml)
+
+##### Outputs
+| Output             | Description
+|--------------------|----------------------------------------------
+| `<stack-group>`    | Detected stack buildpack group  (see [`group.toml`](#grouptoml-toml))
+
+#### `builder`
+The platform MUST execute `builder` in the **build environment**
+
+If the `<stack-group>` IS provided, the `builder` MUST be run with root privileges. If the the `<stack-group>` IS NOT provided, the `builder` MUST NOT be run with root privileges.
+
+Usage:
+```
+/cnb/lifecycle/builder \
+  ...
+  [-stack-buildpacks <stack-buildpacks>] \
+  [-stack-group <stack-group>]
+```
+
+##### Inputs
+| Input                 | Environment Variable        | Default Value           | Description
+|-----------------------|-----------------------------|-------------------------|----------------------
+| ...                   |                             |                         |
+| `<stack-buildpacks>`  | `CNB_STACK_BUILDPACKS_DIR`  | `/cnb/stack/buildpacks` | Path to stack buildpacks directory (see [Buildpacks Directory Layout]
+| `<stack-group>`       | `CNB_STACK_GROUP_PATH`      | `./stack-group.toml`    | Path to output group definition(#buildpacks-directory-layout))
+
+##### Outputs
+| Output                                     | Description
```

Comment:
We should add `<layers>/<buildpack ID>/stack-layer.toml` to the output list.

Reply:
Fixed in eb9cb5d

Reply:
Oh, but I think it should be `<layers>/stack-layer.toml` as `<buildpack ID>` is included in `<layers>`